example/src/App.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+  Button,
+  ScrollView,
+  Platform,
+} from 'react-native';
+import TeadsAdView from '../../src/teads-ad-view';
+import Teads from '../../src/teads';
+import TeadsAdPlacementSettings from '../../src/teads-ad-placement-settings';
+import TeadsAdRequestSettings from '../../src/teads-ad-request-settings';
+import TeadsAdRatio from '../../src/teads-ad-ratio';
+import { Dimensions } from 'react-native';
+import TeadsInReadAdPlacement from '../../src/teads-inread-ad-placement';
+import { DeviceEventEmitter } from 'react-native';
+//import TeadsAdViewEvents from '../../src/teads-ad-view-events';
+import { NativeEventEmitter, NativeModules } from 'react-native';
+
+export default function App() {
+  //const testTeadsAdViewEvents = new TeadsAdViewEvents();
+
+  const [showAd, setShowAd] = React.useState<boolean>(true);
+  const [adId, setAdId] = React.useState<String>();
+  const [height, setHeight] = React.useState<number>();
+  const [placement, setPlacement] = React.useState<
+    TeadsInReadAdPlacement | undefined
+  >();
+
+  const testAdPlacementSetting = new TeadsAdPlacementSettings();
+  const testAdRequestSettings = new TeadsAdRequestSettings();
+  const testAdRatio = new TeadsAdRatio(Dimensions.get('window').width);
+
+  React.useEffect(() => {
+    (async () => {
+      await testAdPlacementSetting.enableDebug();
+      await testAdRequestSettings.enableValidationMode();
+      const awaitedVal = await Teads.createInReadPlacement(
+        84242,
+        testAdPlacementSetting
+      );
+      setPlacement(awaitedVal);
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  async function onPress() {
+    const resultId = await placement?.requestAd(testAdRequestSettings);
+    setAdId(resultId);
+
+    if (resultId) {
+      testAdRatio.calculateHeight(testAdRatio.width, resultId).then(setHeight);
+    }
+  }
+
+  async function toggleDisplay() {
+    setShowAd(!showAd);
+  }
+
+  async function getEvents() {
+    if (Platform.OS === 'ios') {
```

Comment:
I don't like the fact that there is a specific implementation for each platform
Also, you are receiving events on a kind of singleton meaning if there are multiple teads ads on the same page, it's hard to know which one of the ads is getting an impression event.
I think you should try another approach that could relate an ad to events

Reply:
<img width="564" alt="Capture d’écran 2023-06-12 à 15 00 29" src="https://github.com/teads/TeadsSDK-ReactNative/assets/99195946/c1c2be5d-a90e-4fe6-8676-06bf50aec872">